File: src/main/java/kr/hahaha98757/zombiesaddon/commands/CommandZombiesAddon.java
package kr.hahaha98757.zombiesaddon.commands;

import kr.hahaha98757.zombiesaddon.UpdateChecker;
import kr.hahaha98757.zombiesaddon.ZombiesAddon;
import kr.hahaha98757.zombiesaddon.hudposition.ConfigGui;
import kr.hahaha98757.zombiesaddon.hudposition.DelayedTask;
import kr.hahaha98757.zombiesaddon.utils.Utils;
import net.minecraft.client.Minecraft;
import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.command.WrongUsageException;
import net.minecraft.util.BlockPos;

import java.util.List;

public class CommandZombiesAddon extends CommandBase {

    @Override
    public String getCommandName() {
        return ZombiesAddon.MODID;
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 0;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "/zombiesaddon <hud|checkUpdate|updateLog>";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) throws CommandException {
        if (sender != Minecraft.getMinecraft().thePlayer) return;

        if (args.length == 0) throw new WrongUsageException(getCommandUsage(null));

        switch (args[0]) {
            case "hud":
                new DelayedTask(() -> Minecraft.getMinecraft().displayGuiScreen(new ConfigGui()), 2);
                break;
            case "checkUpdate":
                Utils.addTranslationChat("zombiesaddon.commands.zombiesaddon.checkingUpdate", new Object());
                UpdateChecker.checkUpdate();
                Utils.addTranslationChat("zombiesaddon.commands.zombiesaddon.checkedUpdate", new Object());
                break;
            case "updateLog":
                Utils.addChatWithURL(Utils.LINE + "\n", "§9§nClick here to view log.", "https://github.com/hahaha98757/zombies-addon?tab=readme-ov-file#update-log", "Open URL.", "\n" + Utils.LINE);
                break;
            default:
                throw new WrongUsageException(getCommandUsage(null));
        }
    }

    @Override
    public List<String> addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos) {
        if (args.length == 1) return getListOfStringsMatchingLastWord(args, "hud", "checkUpdate", "updateLog");
        return null;
    }
}

File: src/main/kotlin/kr/hahaha98757/zombiesaddon/commands/CommandZombiesAddon.kt
package kr.hahaha98757.zombiesaddon.commands

import kr.hahaha98757.zombiesaddon.MODID
import kr.hahaha98757.zombiesaddon.ZombiesAddon
import kr.hahaha98757.zombiesaddon.enums.Difficulty
import kr.hahaha98757.zombiesaddon.gui.hudposition.ConfigGui
import kr.hahaha98757.zombiesaddon.update.UpdateChecker
import kr.hahaha98757.zombiesaddon.utils.DelayedTask
import kr.hahaha98757.zombiesaddon.utils.addTranslatedChat
import kr.hahaha98757.zombiesaddon.utils.isNotPlayZombies
import kr.hahaha98757.zombiesaddon.utils.mc
import net.minecraft.command.ICommandSender
import net.minecraft.command.WrongUsageException
import net.minecraft.util.BlockPos

object CommandZombiesAddon: CustomCommandBase() {
    override fun getCommandName() = MODID
    override fun getCommandUsage(sender: ICommandSender?) = "/zombiesaddon <difficulty|hud|checkUpdate>"
    override fun runCommand(sender: ICommandSender, args: Array<String>) {
        if (args.isEmpty()) throw WrongUsageException(getCommandUsage(null))
        when (args[0]) {
            "difficulty" -> {
                if (args.size < 2) throw WrongUsageException("/zombiesaddon difficulty <normal|hard|rip>")
                if (isNotPlayZombies()) {
                    addTranslatedChat("zombiesaddon.commands.zombiesaddon.difficulty.failed")
                    return
                }
                val game = ZombiesAddon.instance.gameManager.game ?: return
                val difficulty = when (args[1]) {
                    "normal" -> {
                        game.changeDifficulty(Difficulty.NORMAL)
                        "§aNormal"
                    }
                    "hard" -> {
                        game.changeDifficulty(Difficulty.HARD)
                        "§cHard"
                    }
                    "rip" -> {
                        game.changeDifficulty(Difficulty.RIP)
                        "§4RIP"
                    }
                    else -> throw WrongUsageException("/zombiesaddon difficulty <normal|hard|rip>")
                }
                addTranslatedChat("zombiesaddon.commands.zombiesaddon.difficulty", difficulty)
            }
            "hud" -> DelayedTask { mc.displayGuiScreen(ConfigGui()) }
            "checkUpdate" -> UpdateChecker.initAndCheck()
            else -> throw WrongUsageException(getCommandUsage(null))
        }
    }

    override fun addTabCompletionOptions(sender: ICommandSender?, args: Array<String?>, pos: BlockPos?): List<String>? =
        if (args.size == 1) getListOfStringsMatchingLastWord(args, "difficulty", "hud", "checkUpdate")
        else if (args.size == 2 && args[0] == "difficulty") getListOfStringsMatchingLastWord(args, "normal", "hard", "rip")
        else null
}